sklearn/decomposition/sparse_pca.py

```diff
@@ -135,6 +143,8 @@ def fit(self, X, y=None):
                                                )
         self.components_ = Vt.T
         self.error_ = E
+        if self.variance:
+            self.explained_variance_ = self._get_explained_variance(X)
```

Comment:
sorry one more thing. You should add
```python
else:
    self.explained_variance_ = None
```
otherwise you'd get an attribute error if calling estimator.explained_variance afterwards. 

and you can add a test which checks that when variance is False, explained_variance_ is not computed.

Reply:
added thanks! one random question, i see in your PR, you added to "doc/whats_new/v0.20.rst"
should I be adding the same to a similar file? (and which?)

Reply:
Yes you'll have to add it in the what's new. But I don't know if your PR will be in the 0.20 release. Let it as is for now and we'll fill the right what's new as soon as we know.